planning/behavior_velocity_crosswalk_module/src/scene_crosswalk.cpp

```diff
@@ -241,7 +241,8 @@ bool CrosswalkModule::modifyPathVelocity(PathWithLaneId * path, StopReason * sto
     return t && cmp_fn((now - *t).seconds(), planner_param_.occlusion_time_buffer);
   };
   const auto is_crosswalk_ignored =
-    (planner_param_.occlusion_ignore_with_red_traffic_light && isRedSignalForPedestrians()) ||
+    (planner_param_.occlusion_ignore_with_traffic_light &&
+     !road_.regulatoryElementsAs<const lanelet::TrafficLight>().empty()) ||
```

Comment:
maybe create an inline `_roadHasTrafficLight()` for this to make more readable?

Reply:
also looks like `isRedSignalForPedestrians` is not used anymore.

Reply:
Thank you for your review.
`isRedSignalForPedestrians` is used somewhere else so I cannot remove it.
I made an intermediate variable `crosswalk_has_traffic_light` to make the condition more readable. I also fixed a bug (we care about the crosswalk's traffic light, not the road's).
[0bebece](https://github.com/autowarefoundation/autoware.universe/pull/6604/commits/0bebece9a62eeb94e56aa6b06d63fea31514e722)